01-node-tutorial/answers/12-http.js

```diff
@@ -0,0 +1,17 @@
+const http = require('http');
+
+const server = http.createServer((req,res) =>{
+    if (req.url === '/'){
+        res.end('Welcome to our home page')
+    }
+    if (req.url === '/about'){
+        res.end('Here is our short history')
+    }
+    res.end(`
+    <h1>Oops!</h1>
+    <p>We can't seem to find the page you are looking for</p>
+    <a href="/">back home</a>
+    `)
+})
```

Comment:
```suggestion
    if (req.url === '/'){
        res.end('Welcome to our home page')
        return;
    }
    if (req.url === '/about'){
        res.end('Here is our short history')
        return;
    }
    res.end(`
    <h1>Oops!</h1>
    <p>We can't seem to find the page you are looking for</p>
    <a href="/">back home</a>
    `)
})
```

A bug can occur if you call `res.end` more than once. It doesn't always happen, it's one of those erroneous inconsistent bugs. You can fix it by making sure the `end` function is only called once by adding a return statement after every call to `res.end`. You can also solve this problem with a chain of `if`, `if else`, and `else` statements but that fix is a bit tricker